.travis/build-coveralls.sh

```diff
@@ -4,7 +4,7 @@ set -e
 source .travis/logger.sh
 source .travis/common-functions.sh
 
-VERSION=$(./mvnw -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive org.codehaus.mojo:exec-maven-plugin:1.5.0:exec)
+VERSION=$(./mvnw -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive org.codehaus.mojo:exec-maven-plugin:3.0.0:exec)
```

Comment:
It looks like this should be refactored and reused. We could define a function in `.travis/common-functions.sh` - then we have in the future only one place, where we need to update the plugin version.

On a side note: The goal of this is, to figure out, which version we are currently building. Calling the exec-maven-plugin is one way to do this, which works, but it's very heavy (starting maven to only read the `<version>` element of the pom.xml file....). Maybe you know a better way to read/parse the version from the file in the shell?

Reply:
It's indeed a good idea to define a function in common-functions.sh 👌 

However it sounds easier (more lightweight) to read pom.xml and find project.version from there, wdyt? I see it either as a maven exec or maven help plugin OR bash + some scripting for parsing pom.

Most of the comments are related to [maven-help-plugin](https://maven.apache.org/plugins/maven-help-plugin/evaluate-mojo.html) but it means changing one maven call to another one

Reply:
I've found this page: https://html.developreference.com/article/20765025/How+to+get+Maven+groupId%2C+artifactId+%26+project+version+to+the+command+line

The shell script variant looks like this:
```
VERSION=$(echo -e 'setns x=http://maven.apache.org/POM/4.0.0\ncat /x:project/x:version/text()' | xmllint --shell pom.xml | grep -v /)
```

It uses xmllint, but I don't know, whether this is available on travis. You could try this in one commit and see, whether the build still works. Or we just stay with maven here...

Reply:
ok, xmlint is not available:

https://travis-ci.org/github/pmd/pmd/jobs/708823277#L280

I'll revert my change and merge this PR then.

Reply:
I wanted to finish this but was a bit overwhelmed with daily work :/

>It uses xmllint, but I don't know, whether this is available on travis. You could try this in one commit and see, whether the build still works. Or we just stay with maven here...

IMO there was no use of changing one crutch by another one :) So staying with exec-maven-plugin is not a bad idea.